src/content/tutorials/fleet-management/cluster-management/cluster-upgrades/index.md

```diff
@@ -35,152 +31,71 @@ The workload cluster stack includes third-party components such as:
 
-**Patch and minor upgrades** are rolled out automatically by Giant Swarm, respecting agreed maintenance windows. Our team is developing the automation necessary to further streamline this process, allowing customers to manage maintenance windows effectively.
+You cannot skip major versions when upgrading - upgrades must proceed sequentially through major versions. You can skip minor and patch releases if needed.
 
-**Major upgrades** are not automated. You are notified to schedule the upgrade, possibly after updating workloads. Giant Swarm staff may assist or initiate these upgrades to ensure continuity.
+To trigger an upgrade:
 
-### Skipping workload cluster release
+- **GitOps**: Update the cluster configuration file with the desired release version
+- **kubectl-gs**: Use the [`update cluster`]({{< relref "/reference/kubectl-gs/update-cluster" >}}) command with the `--release-version` flag
 
-Skipping major versions when upgrading is unsupported. Upgrades must proceed sequentially through major versions.
+## Avoid downtime during upgrades
 
-You can skip minor and patch releases if needed. Our interfaces default to the next active workload cluster release. To skip a version:
+To ensure your workloads remain operational during upgrades:
 
-- In the **GitOps** workflow, update the cluster configuration file with the desired release version.
-- Using **kubectl-gs**, use the [`update cluster`]({{< relref "/reference/kubectl-gs/update-cluster" >}}) command with the `--release-version` flag.
+- **Ensure redundancy**: Use two or more replicas for all critical deployments
+- **Distribute workloads**: Use [inter-pod anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) to spread replicas across nodes
+- **Use probes**: Implement [liveness and readiness probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/) with appropriate `initialDelaySeconds` values
+- **Handle termination signals**: Ensure pods handle `TERM` signals gracefully and configure `terminationGracePeriodSeconds` for longer shutdowns if needed
+- **Set disruption budgets**: Use [PodDisruptionBudgets](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) to maintain minimum pod counts (avoid `maxUnavailable=0` for single-replica deployments)
+- **Set scheduling priorities**: Use Pod priority classes and set resource requests/limits to influence scheduling
+- **Avoid ephemeral resources**: Use controllers (Deployments, StatefulSets) instead of standalone Pods
+- **Configure webhook timeouts**: Set low timeout values (e.g., 5 seconds) for validation and mutation webhooks
+- **Verify pod health**: Ensure all critical pods are running before upgrades with `kubectl get pods --all-namespaces | grep -v "Running\|ContainerCreating"`
```

Comment:
```suggestion
- **Configure webhook timeouts**: Set low timeout values (e.g., 5 seconds) for validation and mutation webhooks
- **Verify pod health**: Ensure all critical pods are running before upgrades with `kubectl get pods --all-namespaces | grep -v "Running\|Completed"`
```

`ContainerCreating` is similar to `Pending`, so only `Running` and `Completed` are _good_ signals.